packages/@uppy/aws-s3-multipart/src/index.js

```diff
@@ -430,6 +434,56 @@ export default class AwsS3Multipart extends BasePlugin {
       .then(assertServerError)
   }
 
+  /**
+   * @type {import("../types").AwsS3STSResponse | Promise<import("../types").AwsS3STSResponse>}
+   */
+  #cachedTemporaryCredentials
+
+  async #getTemporarySecurityCredentials (options) {
+    throwIfAborted(options?.signal)
+
+    if (this.#cachedTemporaryCredentials == null) {
+      // We do not await it just yet, so concurrent calls do not try to override it:
+      if (this.opts.getTemporarySecurityCredentials === true) {
+        this.assertHost('getTemporarySecurityCredentials')
+        this.#cachedTemporaryCredentials = this.#client.get('s3/sts', null, options).then(assertServerError)
+      } else {
+        this.#cachedTemporaryCredentials = this.opts.getTemporarySecurityCredentials(options)
+      }
+      setTimeout(() => {
+        this.#cachedTemporaryCredentials = null
+      }, (await this.#cachedTemporaryCredentials).credentials.expires)
```

Comment:
I ended up using the HTTP cache for that, it seems the be the most efficient way to share state between tabs/Uppy instances. Now I'm not sure how the client can tell how long until the credentials expire – which would be nice to have because if it expires in the middle of an upload, the uploading chunks could abort immediately, otherwise the client has to wait until the full chunk is uploaded and the remote server tells it it's expired.